### PR TITLE
Update button

### LINKED ISSUE #4407: Indicate that there is a new modeler version available
### Problem you would like to solve

As a user I get feedback when I have a freshly (new) installed modeler version (via the what's new communcation). However, I do not get feedback when there is a new version of the modeler available, i.e. I skipped the update check in the past. This allows me to forget about the upgrade.

What I'd like to see is that the modeler nudges me more into upgrading, i.e. by indicating in the what's new area that a new version of the modeler is available.

### Proposed solution

* [ ] What's new indicates via a :red_circle: that a new version is available (you're running on an outdated version!)
* [ ] Clicking the button does not show what's new, but a call to upgrade the modeler

### Alternatives considered

Auto-upgrade, which (as we discussed previously) is not always possible.

### Additional context

Example how an upgrade nudge could look like (visually):

image

Clicking the version opens an update notice:

image

Clicking `Learn what's new` opens the update notice.

### PR DESCRIPTION
**Pull request by abdul99ahad**
Closes #4407

### Proposed Changes

 

image

image

Clicking `Learn what's new` will open modal with release changes 

image

Clicking `Update Now` will take you to downloads page. 

Note: Release notes are not visible because the update behavior is being mocked.

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] __Brief textual description__ of the changes present
* [x] __Visual demo__ attached
* [ ] __Steps to try out__ present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: client/src/plugins/report-feedback/ReportFeedback.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ export class ReportFeedback extends PureComponent {
 
     return (
       <Fragment>
-        <Fill slot="status-bar__app" group="9_feedback">
+        <Fill slot="status-bar__app" group="8_feedback">
```
Comment (nikku) [inline client/src/plugins/report-feedback/ReportFeedback.js:103]: What is the reason for this change? It moves buttons around (on the status bar).

Comment (abdul99ahad) [reply client/src/plugins/report-feedback/ReportFeedback.js:103]: To accomodate the new `Update` button

**Comment by lmbateman** [comment]
Visually it looks fine to me. Two minor points on the text:

- The button says "Update", but the popup says "Upgrade". We should use the same word everywhere. If there's a difference between "update" and "upgrade", we should use the appropriate word (which I suspect is "update").
- The [Camunda style guide](https://confluence.camunda.com/display/HAN/Camunda+style+guide) specifies sentence case, so in the popup, "Update Now" should be "Update now".

Thanks!

File: client/src/plugins/update-checks/UpdateAvailableOverlay.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  );
+}
+
+function UpdateAvailableSection({ version, openVersionInfoPage, onGoToDownloadPage }) {
+
+  return (
+    <div className={ css.UpdateAvailableOverlay }>
+      <Section maxHeight="500px">
+        <Section.Header>
+          Upgrade available
```
Comment (philippfromme) [inline client/src/plugins/update-checks/UpdateAvailableOverlay.js:36]:
```suggestion
          Update available
```

File: client/src/plugins/update-checks/UpdateAvailableOverlay.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        </Section.Header>
+        <Section.Body>
+          <p>Camunda Desktop Modeler {version} is available for use.</p>
+          <a className="links" onClick={ onGoToDownloadPage }> Update Now </a>
```
Comment (philippfromme) [inline client/src/plugins/update-checks/UpdateAvailableOverlay.js:40]:
```suggestion
          <a className="links" onClick={ onGoToDownloadPage }>Update now</a>
```

File: client/src/plugins/update-checks/UpdateChecks.js [hunk 4]
```diff
@@ -335,16 +366,49 @@ export default class UpdateChecks extends PureComponent {
...
+      toggle,
+      _buttonRef: buttonRef,
+      close,
+      openVersionInfoPage,
+    } = this;
     return (
       <React.Fragment>
+        {updateAvailable && <Fill slot="status-bar__app" group="9_update_checks">
+          <button
+            style={ {
```
Comment (philippfromme) [inline client/src/plugins/update-checks/UpdateChecks.js:395]: Why are we using inline styles here?

File: client/src/plugins/update-checks/UpdateChecks.js [hunk 5]
```diff
@@ -335,16 +366,49 @@ export default class UpdateChecks extends PureComponent {
...
+              backgroundColor: 'rgba(0, 119, 204, 1)',
+              color: 'rgb(254, 254, 254)'
+            } }
+            className="btn btn-primary"
+            title="Toggle update info"
+            onClick={ toggle }
+            ref={ buttonRef }
+          >{ 'Update' } </button>
+        </Fill>}
+        {open && <UpdateAvailableOverlay
```
Comment (philippfromme) [inline client/src/plugins/update-checks/UpdateChecks.js:405]: Consider renaming `open` to clarify what is open as this component handles multiple things.

File: client/src/plugins/update-checks/UpdateChecks.js (hunk 4 above)
Comment (abdul99ahad) [reply client/src/plugins/update-checks/UpdateChecks.js:395]: The footer element styles are inline (talking about Saved button and 💚 icon). I tried creating .less file to add styles initially, but it wasn't working so by looking at the reference buttons, I assumed it's not supposed to work this way. If we can eliminate the inline styles and use separate files for styles in Fill component, then we have to do it across all elements for consistency.

**Comment by abdul99ahad** [comment]
Ready for review

File: client/src/plugins/update-checks/UpdateChecks.js (hunk 4 above)
Comment (philippfromme) [reply client/src/plugins/update-checks/UpdateChecks.js:395]: We could simply add a `btn--primary` class to `StatusBar.less` and use it for the button.

**Comment by philippfromme** [comment]
Mac is finally passing. Will do the final review now.

**Review by philippfromme — APPROVED** [review]
I did a couple of small adjustments:

* simplified CSS
* adjusted naming for clarity
* fixed whitespace
